presto-main/src/main/resources/webapp/src/components/SQLClient.jsx

```diff
@@ -275,55 +294,247 @@ function QueryResults({results}) {
+    const [values, setValues] = React.useState({sql: '', running: false, results: undefined, view: 'SQL'});
+    const sessions: SessionValues = React.useRef({});
+    const views = [{name: 'SQL', label: 'SQL'}, {name: 'Session', label: 'Session Properties'}];
 
     const executeSQL = (sqlInfo) => {
-        const client = createClient(sqlInfo.catalog, sqlInfo.schema);
+        const client = createClient(
+            sqlInfo.catalog,
+            sqlInfo.schema,
+            Object.keys(sessions.current).map(k => `${k}=${sessions.current[k]}`).join(', ')
+        );
         setValues({ ...values, running: true, results: undefined });
-        client.query(sqlInfo.sql).then((prestoQuery) => {
-            if (prestoQuery.data && prestoQuery.data.length > 0) {
-                setValues({ ...values, running: false, results: prestoQuery });
-            } else {
-                // presto-js-client doesn't return error
-                // need to query the error on our own
-                fetch('/v1/query/' + prestoQuery.queryId)
-                    .then(response => response.json())
-                    .then(query => {
-                        if (query.errorCode) {
-                            setValues({
-                                ...values, running: false, results: {
-                                    error: {
-                                        code: query.errorCode.name,
-                                        message: query.failureInfo.message,
-                                    }
-                                }
-                            });
-                        } else {
-                            // the query returns zero record
-                            setValues({ ...values, running: false, results: prestoQuery });
-                        }
-                    })
-                    .catch((error) => {
-                        setValues({ ...values, running: false, results: { error } });
-                    });
-            }
+        client.query(sqlInfo.sql).then((prestoQuery: PrestoQuery) => {
+            setValues({ ...values, running: false, results: prestoQuery });
         })
         .catch((e) => {
-            // presto-js-client only return error type in the error message
-            // it would be good to also have detailed message
             setValues({...values, results: {error: e}});
         });
     };
 
+    const switchView = (view) => {
+        if (view.name === values.view) return;
+        setValues({...values, view: view.name});
+    };
+
     const handleError = (err) => {
         setValues({...values, results: {error: {message: err.message}}});
+    };
+
+    const sessionHandler = (row) => {
+        if (sessions.current[row[0]] && row[1] === row[2]) {
+            // revert the change back to default
+            delete sessions.current[row[0]];
+        } else {
+            sessions.current[row[0]] = row[1];
+        }
```

Comment:
```suggestion
    const sessionHandler = ([name, currentValue, defaultValue]) => {
        if (sessions.current[name] && currentValue === defaultValue) {
            // revert the change back to default
            delete sessions.current[name];
        } else {
            sessions.current[name] = currentValue;
        }
```
Consider using array destructuring in favor of more readable code. I'm not sure if I got the name of the positions right, I guessed them based on the object on line 420 in this file.

Reply:
I used `name`, `value`, and `defaultValue` :-)